WcaOnRails/app/webpacker/components/Panel/Board/RegionManager.jsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import {
+  Button, ButtonGroup, Confirm, Form, Header, List, Modal,
+} from 'semantic-ui-react';
+import useLoadedData from '../../../lib/hooks/useLoadedData';
+import { userGroupsUrl, userGroupsUpdateUrl } from '../../../lib/requests/routes.js.erb';
+import Errored from '../../Requests/Errored';
+import Loading from '../../Requests/Loading';
+import useSaveAction from '../../../lib/hooks/useSaveAction';
+
+const defaultRegion = {
+  name: '',
+  group_type: 'delegate_regions',
+  parent_group_id: null,
+  is_active: true,
+  is_hidden: false,
+};
+
+function UserGroupVisibility({ userGroup, save, sync }) {
+  const [open, setOpen] = React.useState(false);
+  return (
+    <>
+      {!userGroup.is_hidden && (
+      <List.Icon
+        name="eye"
+        link
+        onClick={() => setOpen(true)}
+      />
+      )}
+      {userGroup.is_hidden && (
+      <List.Icon
+        name="eye slash"
+        link
+        onClick={() => setOpen(true)}
+      />
+      )}
+      <Confirm
+        open={open}
+        content={`Are you sure you want to ${userGroup.is_hidden ? 'unhide' : 'hide'} ${userGroup.name}?`}
+        onCancel={() => setOpen(false)}
+        onConfirm={() => {
+          setOpen(false);
+          save(
+            userGroupsUpdateUrl(userGroup.id),
+            { is_hidden: !userGroup.is_hidden },
+            sync,
+          );
+        }}
+      />
+    </>
+  );
+}
+
+export default function RegionManager() {
+  const {
+    data, loading, error, sync,
+  } = useLoadedData(userGroupsUrl({
+    groupType: 'delegate_regions',
+  }));
+  const { save, saving } = useSaveAction();
+  const [openNewRegionModal, setOpenNewRegionModal] = React.useState(false);
+  const [openNewSubregionModal, setOpenNewSubregionModal] = React.useState(false);
```

Comment:
Since there's just a single modal reused between these, it might be simpler to track the type of modal
```
const [openModalType, setOpenModalType] = useState(null) // options: 'region' and 'subregion'
```
This reduces the number of states, and makes it clear that you can't have both modals open at once. But the current implementation is also fine, so feel free to ignore this.

Reply:
Done.